source/posix/clock.c

```diff
@@ -9,7 +9,9 @@
 
 static const uint64_t NS_PER_SEC = 1000000000;
 
-#if defined(CLOCK_MONOTONIC_RAW)
+#if defined(CLOCK_BOOTTIME)
+#    define HIGH_RES_CLOCK CLOCK_BOOTTIME
+#elif defined(CLOCK_MONOTONIC_RAW)
 #    define HIGH_RES_CLOCK CLOCK_MONOTONIC_RAW
```

Comment:
Previously we preferred CLOCK_MONOTONIC_RAW:
[docs](https://linux.die.net/man/2/clock_gettime) say:
> Similar to CLOCK_MONOTONIC, but provides access to a raw hardware-based time that is not subject to NTP adjustments or the incremental adjustments performed by adjtime.

but with this change we'd prefer CLOCK_BOOTTIME, which days say:
> Identical to CLOCK_MONOTONIC, except it also includes any time that the system is suspended.

So we'd be subject again to "NTP adjustments or the incremental adjustments performed by adjtime"

Just pointing it out. I don't know enough about this off the top of my head to gauge the full repercussions

Reply:
OK. I just did some more reading on all this. BOOTTIME seems fine for our purposes.